pkg/ccl/changefeedccl/enriched_source_provider.go

```diff
@@ -371,3 +500,34 @@ func init() {
 }
 
 const originCockroachDB = "cockroachdb"
+
+func getDescriptors(
+	ctx context.Context, execCfg *sql.ExecutorConfig, tableID descpb.ID,
+) (catalog.TableDescriptor, catalog.DatabaseDescriptor, catalog.SchemaDescriptor, error) {
+	var tableDescriptor catalog.TableDescriptor
+	var dbDescriptor catalog.DatabaseDescriptor
+	var schemaDescriptor catalog.SchemaDescriptor
+	f := func(ctx context.Context, txn descs.Txn) error {
+		byIDGetter := txn.Descriptors().ByIDWithoutLeased(txn.KV()).WithoutNonPublic().Get()
+		tableDesc, err := byIDGetter.Table(ctx, tableID)
+		if err != nil {
+			return err
+		}
+		dbDesc, err := byIDGetter.Database(ctx, tableDesc.GetParentID())
+		if err != nil {
+			return err
+		}
+		schemaDesc, err := byIDGetter.Schema(ctx, tableDesc.GetParentSchemaID())
+		if err != nil {
+			return err
+		}
+		tableDescriptor = tableDesc
+		dbDescriptor = dbDesc
+		schemaDescriptor = schemaDesc
+		return nil
+	}
+	if err := execCfg.InternalDB.DescsTxn(ctx, f, isql.WithPriority(admissionpb.LowPri)); err != nil {
```

Comment:
is this low priority? since this is required for the feed to start up and only run once, maybe it should be normal priority